tests/test_upload.py

```diff
@@ -383,12 +383,19 @@ def test_compose_file_mapping(tmpdir, monkeypatch):
     monkeypatch.setattr("os.path.isfile", isfile)
 
     # mock gsutil ls file check by returning input
-    monkeypatch.setattr(
-        "subprocess.run", lambda args, capture_output: MagicMock(returncode=0)
+    ls_return = MagicMock()
+    ls_return.returncode = 0
+    ls_return.stdout = "gs://bucket:\ngs://bucket/gcs.path\ngs://bucket/[brackets]/subitem".encode(
```

Comment:
Does this example data test for handling empty lines in `gsutil ls -r` output?

Reply:
add empty line in next commit